### PR TITLE
feat: Add `IsDisplayNameDupeOfCommunityMember ` endpoint

### LINKED ISSUE #13434: [Profile Showcase - Identity] Validate the display name against all members of joined communities
### Description

Figma: https://www.figma.com/file/ibJOTPlNtIxESwS96vJb06/👤-Profile-%7C-Desktop?type=design&node-id=2460-27121&mode=design&t=ZmeSQiEF0IUZft3G-0

There is a new validation requirement when setting up a display name. This validation states that the user cannot choose a display name that is identical to any member of his joined communities.

The purpose of this task is to implement the backend support for this validation.

Screenshot 2024-02-06 at 14 58 01

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #13434
Waits https://github.com/status-im/status-go/pull/4839

### What does the PR do

 Validate the display name against all members of joined communities

### Affected areas

Communities

### CODE REVIEW

**Review by mprakhov — APPROVED** [review]
LGTM, I have only one question:
Will it be ok, if we will join the community, which has user with the same name?

**Comment by MishkaRogachev** [comment]
> LGTM, I have only one question: Will it be ok, if we will join the community, which has user with the same name?

As far as I know, the decision is to limit it to current communities for now.

**Review by jrainville — APPROVED** [review]

File: ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml [hunk 1]
```diff
@@ -100,6 +100,10 @@ QtObject {
         return root.communitiesModuleInst.getCommunityPublicKeyFromPrivateKey(privateKey);
     }
 
+    function scanCommunitiesMembersForName(displayName) {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml:103]: Could use a better name... it doesn't really tell the QML user that this returns a boolean and moreover what's the meaning of it

Comment (MishkaRogachev) [reply ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml:103]: `isDisplayNameOccursAmongCommunitiesMembers`?

Comment (caybro) [reply ui/app/AppLayouts/Communities/stores/CommunitiesStore.qml]: OK, then what about `isDisplayNameDupeOfCommunityMember`?

**Pull request merged**